src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -38,4 +38,20 @@ public BookDto findById(long id) {
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto createBookRequestDto) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Book not found by id: " + id));
```

Comment:
we should also add this validation for the delete method, to explicitly say that it was or was not deleted. So let's extract this part to a private method and reuse it